### ISSUE TITLE
Volume quota not being applied for all clients

### ISSUE THREAD

**Issue by cricket1** [openNDS/openNDS#627]
I have set the   Download quota to 5000 kB and   Upload quota to 5000 kB globally.I dont see this quota being applied to all clients.

Here is the output for 3 clients.Client 0 the quota is set but not for other clients.Is there something wrong with the configuration

openNDS Status
====
Version: 10.3.0
Uptime: 6m 53s
Gateway Name: [ openNDS Node:f0090d2dc042  ]
Debug Level: [ 1 ]
Gateway FQDN: [ status.client ]
Managed interface: phy0-ap0
Upstream gateway(s) [ online:192.168.2.1,wan  ]
MHD Server [ version 0.9.75 ] listening on: http://10.10.20.3:2050
Maximum Html Page size is [ 10240 ] Bytes
Preemptive Authentication is Enabled
Binauth Script: /usr/lib/opennds/binauth_log.sh
ThemeSpec: Disabled
FAS: Secure Level 1, URL: http://138.91.14.166:80/fas-hid.php
Client Check Interval: 15s
Rate Check Window: 2 check intervals (30s)
Preauthenticated Client Idle Timeout: 30m
Authenticated Client Idle Timeout: 120m
Download rate limit threshold (default per client): no limit
Upload rate limit threshold (default per client): no limit
Download quota (default per client): 5000 kB
Upload quota (default per client): 5000 kB
Total download: 23842 kByte; average: 472.92 kbit/s
Total upload: 837 kByte; average: 16.60 kbit/s
====
Client authentications since start: 2
Current clients: 3

Client 0
  Client Type: cpd_can
  IP: 10.10.20.139 MAC: cc:2f:71:12:10:79
  Last Activity: Tue Oct 01 15:22:19 2024 (-1s ago)
  Session Start: Tue Oct 01 15:17:02 2024 (5m 16s ago)
  Session End:   Wed Oct 02 15:17:02 2024 (23h 54m 44s left)
  Token: a0392a1c
  State: Authenticated
  Download Rate Limit Threshold: not set
  Upload Rate Limit Threshold: not set
  Download quota: 5000 kB
  Upload quota: 5000 kB
  Download this session: 3657 kB; Session average: 94.80 kb/s
  Upload this session: 315 kB; Session average: 8.18 kb/s

Client 1
  Client Type: cpd_can
  IP: 10.10.20.203 MAC: 6e:f1:5d:62:e7:da
  Last Activity: Tue Oct 01 15:22:19 2024 (-1s ago)
  Session Start: Tue Oct 01 15:20:55 2024 (1m 23s ago)
  Session End:   Wed Oct 02 15:20:55 2024 (23h 58m 37s left)
  Token: 3a0d2c1f
  State: Authenticated
  Download Rate Limit Threshold: not set
  Upload Rate Limit Threshold: not set
  Download quota: not set
  Upload quota: not set
  Download this session: 20184 kB; Session average: 1992.17 kb/s
  Upload this session: 233 kB; Session average: 23.05 kb/s

Client 2
  Client Type: cpd_can
  IP: 10.10.20.170 MAC: aa:ab:7c:56:58:23
  Last Activity: Tue Oct 01 15:20:41 2024 (1m 37s ago)
  Session Start: -
  Session End:   -
  Token: 5084a128
  State: Preauthenticated
  Download Rate Limit Threshold: not set
  Upload Rate Limit Threshold: not set
  Download quota: not set
  Upload quota: not set
  Download this session: 0 kB; Session average: 0.00 kb/s
  Upload this session: 0 kB; Session average: 0.00 kb/s

====
Trusted MAC addresses:
none
====
Walledgarden FQDNs:
none

Walledgarden Ports:
all
====
Blocklist FQDNs:
none

Blocklist Ports:
all
========

**Comment by bluewavenet** [comment]
@cricket1 

> Is there something wrong with the configuration

You tell me, you seem to be keeping it secret.

It would be useful to know what the hardware and operating system is.....

`Managed interface: phy0-ap0`

It is expected that the managed interface be a bridge......

Please show the output of:
`uci show opennds`

Remember to enclose this output between two sets of three backtics so that it is displayed correctly here.

**Comment by cricket1** [comment]
-  Hardware [https://www.tp-link.com/in/home-networking/wifi-router/archer-ax23/](url)
- software 
```
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='23.05.4'
DISTRIB_REVISION='r24012-d8dd03c46f'
DISTRIB_TARGET='ramips/mt7621'
DISTRIB_ARCH='mipsel_24kc'
DISTRIB_DESCRIPTION='OpenWrt 23.05.4 r24012-d8dd03c46f'
DISTRIB_TAINTS=''
```
- output 
```
opennds.@opennds[0]=opennds
opennds.@opennds[0].gatewayinterface='phy0-ap0'
opennds.@opennds[0].uploadquota='5000'
opennds.@opennds[0].downloadquota='5000'
opennds.@opennds[0].fup_upload_throttle_rate='128'
opennds.@opennds[0].fup_download_throttle_rate='128'
opennds.@opennds[0].fasport='80'
opennds.@opennds[0].fasremoteip='138.91.14.166'
opennds.@opennds[0].faspath='/fas-hid.php'
opennds.@opennds[0].faskey='c775e7b757ede630cd0aa1113bd102661ab38829ca52a6422ab782862f268646'
opennds.@opennds[0].fas_secure_enabled='1'
```

**Comment by bluewavenet** [comment]
@cricket1 
`opennds.@opennds[0].gatewayinterface='phy0-ap0'`

This is invalid for OpenWrt.
The effect will be indeterminate errors. 
You should instead use a bridge interface, typically `br-lan`

There is also a known bug when `fas_secure_enabled` is set to 1, that might also effect volume quotas.
See: issue #609

Regardless of this bug, for an Internet hosted FAS, you should consider fas_secure_enabled 3 or 4

For example see:
https://opennds.readthedocs.io/en/stable/fas.html#fas-hid-https-php
This would use `fas_secure_enabled` set to 4.

**Comment by cricket1** [comment]
Thanks for the response.

Is there a way to apply the captive portal to wifi only and not other interfaces

**Comment by bluewavenet** [comment]
@cricket1 

> Is there a way to apply the captive portal to wifi only and not other interfaces

Yes, but this is more a question of how to configure OpenWrt, not openNDS.
What you are probably looking for is a "guest" network scenario.

You could start by reading this:
https://openwrt.org/docs/guide-user/network/wifi/guestwifi/guest-wlan
